app/Http/Controllers/PurchaseController.php

```diff
@@ -59,86 +60,214 @@ public function purchaseCoin(CoinPurchaseRequest $request): JsonResponse
 
     public function purchaseFilter(PurchaseFilterRequest $request): JsonResponse
     {
-        if (auth()->user()->filters->contains($request->filter_id)) {
+        $user = auth()->user();
+        if ($user->filters->contains($request->filter_id)) {
             return response()->json(['message' => 'Filter already purchased'], 400);
-        } else {
-            DB::beginTransaction();
-            $user = auth()->user();
-            $coin = $user->coin;
-            if ($coin < Price::Filter->getPrice()) {
-                DB::rollBack();
-                return response()->json(['message' => 'Insufficient coin'], 400);
+        }
+
+        DB::beginTransaction();
+        try {
+            $subscription = auth('sanctum')->user()->subscription;
+            $customer_id = $subscription->data['customer_id'] ?? null;
+
+            if (!$customer_id) {
+                return response()->json(['message' => 'Invalid customer ID'], 400);
             }
-            $coin -= Price::Filter->getPrice();
-            $user->update(['coin' => $coin]);
+
+            $response = SubscriptionController::fetchSubscriptionStatus($customer_id);
+            if (!$response['success']) {
+                return response()->json(['message' => 'Subscription validation failed'], 400);
```

Comment:
If subscription is not active, users can still purchase filters if that is paid, if the type is subscription, then the subscription will be required